sdk-extensions/jfr-events/src/main/java/io/opentelemetry/sdk/extension/jfr/JfrSpanProcessor.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.extension.jfr;
+
+import static java.util.Objects.nonNull;
+
+import io.opentelemetry.api.trace.SpanContext;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.trace.ReadWriteSpan;
+import io.opentelemetry.sdk.trace.ReadableSpan;
+import io.opentelemetry.sdk.trace.SpanProcessor;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Span processor to create new JFR events for the Span as they are started, and commit on end.
+ *
+ * <p>NOTE: JfrSpanProcessor must be running synchronously to ensure that duration is correctly
+ * captured.
+ */
+public class JfrSpanProcessor implements SpanProcessor {
+
+  private final Map<SpanContext, SpanEvent> spanEvents = new ConcurrentHashMap<>();
+
+  @Override
+  public void onStart(Context parentContext, ReadWriteSpan span) {
+    if (span.getSpanContext().isValid()) {
+      SpanEvent event = new SpanEvent(span.toSpanData());
+      event.begin();
+      spanEvents.put(span.getSpanContext(), event);
+    }
+  }
+
+  @Override
+  public boolean isStartRequired() {
+    return true;
+  }
+
+  @Override
+  public void onEnd(ReadableSpan rs) {
+    SpanEvent event = spanEvents.remove(rs.getSpanContext());
+    if (nonNull(event) && event.shouldCommit()) {
```

Comment:
```suggestion
    if (event != null && event.shouldCommit()) {
```

Reply:
nonNull is part of the standard JDK API any reason not to use it?

Reply:
Yeah there's a perfectly good literal in Java so we should stick to using it where possible (`nonNull` was added in 1.8, presumably to aid with streams but I don't consider it idiomatic for normal code).